File: 344-reverse-string/344-reverse-string.cpp
class Solution {
public:
    
//     recursive way
    void rev(vector<char>& s, int left , int right){
        if(left>=right)return ;
        swap(s[left++],s[right--]);
        rev(s,left,right);
    }
    
    void reverseString(vector<char>& s) {
        int n= s.size();
        // int l=0,r=n-1;
        // while(l<r){
        //     char temp = s[l];
        //     s[l]=s[r];
        //     s[r]=temp;
        //     l++; r--;
        // }
        
        rev(s,0,n-1);
    }
};

File: 344-reverse-string/344-reverse-string.java
class Solution {
//      using recurssion
    public void swap(char[] s ,int  i ,int j){
        char temp = s[i];
        s[i]=s[j];
        s[j]=temp;
    }
    public void reverse(char[] s,int i , int j){
        if(i>=j) return;
        swap(s,i,j);
        reverse(s,i+1,j-1);
        
    }
    
    public void reverseString(char[] s) {
        // int i =0 , j= s.length-1;
        // while(i<j){
        //    char temp = s[i];
        //     s[i]=s[j];
        //     s[j]=temp;
        //     i++; j--;
        // }
        reverse(s,0,s.length-1);
    }
}